### DISCUSSION TITLE
Any better solution than forceSelect?

### DISCUSSION THREAD

**Discussion by cwang1221** [ui5-community/wdi5#147] [Q&A] [answered]
Hi wdi5 experts,

Recently I'm writing tests with wdi5. During my tests, I need to set `forceSelect: true` in many places. To be honest I don't think it's an elegant solution. 

In my test, I want to modulize as much as possible. So I may have an action function called "iClickOkButtonInDialog". For the first time I click the OK button, it doesn't need to set `forceSelect: true`. But later the dialog is closed and opened again. Then I have to add set the property. Otherwise the `.click` will not work.

So I have to add a param `isFirstTime` in the `iClickOkButtonInDialog` function, to tell it if the `forceSelect` property need to be set or not. And I have to clearly know if a call of the function is the first time or not. This is bad.

Another solution is I always set `forceSelect: true`. But from my observation, the test will be executed very slowly.

I don't know much detail behind it, and I have no good idea to solve it. Just want to ask you, if you could think more about that, if there's other better solution.

Thank you and best regards,
Cong

**Comment by Siolto** [comment, marked as answer]
Hi Cong,

`wdi5` stores references of the controls internally to save runtime. The `forceSelect: true` forces `wdi5` to retrieve the control again from the browser context. This is normally only necessary, when your control was destroyed and recreated in the meantime. 
Is your Dialog maybe destroying itself after `iClickOkButtonInDialog` and recreating when opening again?

Best regards,
Simon

**Reply by cwang1221** [reply]
Hi Simon, the page I'm testing is actually not developed by us. It's hard to say. But without `forceSelect: true` in the second step, it will fail. So I guess it is destroyed. My point is, I really hope there's a magical way, which could allow me to just ignore this `forceSelect` thing in my tests. A wonderful wish :)

**Reply by Siolto** [reply]
Hi Cong,

a Dialog, that ist used multiple times, should not be recreated each and every time again. Maybe you can address this to the developers. If not, the only solution I know so far is to always set `forceSelect: true` which, as already written by you, can lead to performance problems.

@vobu @dominikfeininger any ideas?

Best regards,
Simon

**Reply by vobu** [reply]
what @Siolto says - when the `Dialog` isn't attached to the app lifecycle and/or is recreated (instead of initialized once, then referred to) on every request, no way we can "fix" this in `wdi5` → `forceSelect: true` it is :(
